File: sdks/ruby/lib/cyberarmor/audit.rb
# frozen_string_literal: true

require 'json'
require 'net/http'
require 'uri'
require 'time'

module CyberArmor
  # Thin wrapper around audit event emission.
  # Used internally by Client#emit_audit.
  module Audit
    # Build a standardized audit event hash.
    #
    # @param agent_id   [String]
    # @param tenant_id  [String]
    # @param event_type [String]  e.g. "policy_evaluated", "completion_returned"
    # @param payload    [Hash]    arbitrary event-specific data
    # @return [Hash]
    def self.build_event(agent_id:, tenant_id:, event_type:, payload: {})
      {
        agent_id:    agent_id,
        tenant_id:   tenant_id,
        event_type:  event_type,
        timestamp:   Time.now.utc.iso8601(3),
        sdk_version: CyberArmor::VERSION,
        sdk_lang:    'ruby',
        payload:     payload
      }
    end
  end
end

File: sdks/rust/src/audit.rs
// cyberarmor-sdk/src/audit.rs
//
// Audit event construction helpers.

use chrono::Utc;
use serde_json::{json, Value};

pub(crate) const SDK_VERSION: &str = env!("CARGO_PKG_VERSION");
pub(crate) const SDK_LANG: &str = "rust";

/// Build a standardised audit event `Value` ready to POST to the audit service.
pub fn build_event(
    agent_id: &str,
    tenant_id: &str,
    event_type: &str,
    payload: Value,
) -> Value {
    json!({
        "agent_id":    agent_id,
        "tenant_id":   tenant_id,
        "event_type":  event_type,
        "timestamp":   Utc::now().to_rfc3339_opts(chrono::SecondsFormat::Millis, true),
        "sdk_version": SDK_VERSION,
        "sdk_lang":    SDK_LANG,
        "payload":     payload,
    })
}
